src/components/EtherspotContractTransaction.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useContext, useEffect, useId } from 'react';
+import { ethers } from 'ethers';
+
+// contexts
+import EtherspotBatchContext from '../contexts/EtherspotBatchContext';
+
+// types
+import { IEtherspotContractTransaction } from '../types/EtherspotUi';
+
+interface EtherspotContractTransactionProps extends IEtherspotContractTransaction {
+  children?: React.ReactNode;
+}
+
```

Comment:
Can we add a JSDoc block here to explain this function?

Reply:
added

Reply:
we may need to add to other components/functions too